### ISSUE TITLE
Ability to filter image caption

### ISSUE THREAD

**Issue by kallehauge** [WordPress/gutenberg#7363]
**Is your feature request related to a problem? Please describe.**
I would like to be able to alter the caption text on the initial insert of an image (default image block), like it is possible with ex `image_add_caption_text` in the classic WYSIWYG editor.

_(I guess this could be expanded to alter things like `alt` attribute text etc as well?)_

**Describe the solution you'd like**
I would like a filter or some sort of option to alter and/or override the caption text.
The work it works right now with Gutenberg, if I understand it correctly, then the Media window is loaded with the image and their metadata "on demand" and if one or multiple images are selected then they will get injected directly into the editor, so there is no place to change this before insertion. So the only potential option would be to change the values already when the window is opened but this surely seems like a bad workaround.

What would be really ideal from a plugin/theme developers perspective would be to get access to the data here: https://github.com/WordPress/gutenberg/blob/be346ba68c044b70d7723bd6a1cddef84e941a45/edit-post/hooks/blocks/media-upload/index.js#L177

**Comment by draganescu** [comment]
@youknowriad this issue has been added to the Extensibility project which I see you've closed. Should we remove issues that are still relevant from closed projects?

**Comment by youknowriad** [comment]
Yes, we can remove the project assignment, it's not that important though. Other closed project assignments might be important to keep track of though (other than Extensibilty)

**Comment by vreemt** [comment]
it might be good to add this to https://github.com/WordPress/gutenberg/issues/28923 and possibly get a progress update please & thank you

**Comment by jordesign** [comment]
@kallehauge Does the fact that the caption tool is available in the toolbar resolve this for you? Or is this still something to keep in mind?

<img width="637" alt="Screenshot 2023-07-13 at 4 01 11 pm" src="https://github.com/WordPress/gutenberg/assets/1842363/199a2510-9147-4be5-ae30-dc1c4d997e57">

**Comment by kallehauge** [comment]
Uh! It's been a while @jordesign, but I believe I had a few different instances where dynamically defining the caption during insert made sense (there might be different workarounds for some of them, now).

* I believe one of them was translation related, so if different translations was stored as metadata on the attachment, it wasn't possible to change the string that's being used.
* Another instance I vaguely remember is inserting copyright information (either the attachment owner or a metadata value).

TL;DR: I believe it's still as valid for the same use-cases that the older classic editor `image_add_caption_text` filter satisfies (automating work for the editor and allowing 3. parties to generally modify the value)

**Comment by jordesign** [comment]
Thanks for checking in @kallehauge  - given all that it makes sense to leave it open for now 👍

**Comment by carolinan** [comment]
Hi, captions can now be set with the Block Bindings API, does that solve it?

**Comment by Mamaduka** [comment]
I think we can close this as "won't fix"; the use cases listed above can be solved with block binding, as @carolinan mentioned.

**Comment by Mamaduka** [comment]
@justintadlock, IIRC, you've example of similar attributions via Block Bindings.

**Comment by justintadlock** [comment]
I wrote a tutorial but captions were not yet supported at the time: https://developer.wordpress.org/news/2024/02/introducing-block-bindings-part-1-connecting-custom-fields/

I believe caption support was added in WP 6.9, right? If so, this should solve the issue.

**Comment by carolinan** [comment]
Yes
https://make.wordpress.org/core/2025/11/12/block-bindings-improvements-in-wordpress-6-9/

**Comment by Mamaduka** [comment]
Thanks for sharing the links, @carolinan, @justintadlock!

I'm going to close this as "won't fix"; the mentioned behavior can be implemented via the Block Bindings API.
